lib/Backend/amd64/LowererMDArch.cpp

```diff
@@ -11,6 +11,15 @@ const Js::OpCode LowererMD::MDExtend32Opcode = Js::OpCode::MOVSXD;
 
 extern const IRType RegTypes[RegNumCount];
 
+bool
+LowererMDArch::IsArgSaveRequired(Func *func) {
+    return (func->IsJitInDebugMode() || func->IsGeneratorFunc() ||
+        func->GetJnFunction()->GetIsAsmjsMode() || func->IsLambda() ||
```

Comment:
Is there a reason to not do this for asm.js?  I think that after my change, we would never need to emit them for asm.js (even non-leafs).